services/blockchain-indexer/shared/dataService/business/transactionsEstimateFees.js

```diff
@@ -24,29 +29,62 @@ const regex = require('../../regex');
 const { parseToJSONCompatObj } = require('../../utils/parser');
 const { requestConnector, requestFeeEstimator } = require('../../utils/request');
 
+const SIZE_BYTE_SIGNATURE = 64;
+const SIZE_BYTE_ID = 32;
+
+const OPTIONAL_TRANSACTION_PROPERTIES = Object.freeze({
+	FEE: 'fee',
+	SIGNATURES: 'signatures',
+	ID: 'id',
+});
+
+const mockOptionalProperties = (transaction) => {
+	const missingProps = Object.values(OPTIONAL_TRANSACTION_PROPERTIES)
+		.filter((optionalProp) => !Object.keys(transaction).includes(optionalProp));
+
+	missingProps.forEach(prop => {
+		if (prop === OPTIONAL_TRANSACTION_PROPERTIES.FEE) {
+			transaction.fee = 0;
+		}
+		if (prop === OPTIONAL_TRANSACTION_PROPERTIES.SIGNATURES) {
+			transaction.signatures = [getRandomBytes(SIZE_BYTE_SIGNATURE).toString('hex')];
+		}
+		if (prop === OPTIONAL_TRANSACTION_PROPERTIES.ID) {
+			transaction.id = getRandomBytes(SIZE_BYTE_ID).toString('hex');
+		}
+	});
+
+	return transaction;
+};
```

Comment:
Alternative way to write. This way we can optimize into one loop and avoid in place hard coding of value.

```suggestion
const OPTIONAL_TRANSACTION_PROPERTIES = Object.freeze({
	FEE: {
        propName: 'fee',
        defaultValue: 0,
    },
	SIGNATURES: {
        propName: 'signatures',
        defaultValue: [getRandomBytes(SIZE_BYTE_SIGNATURE).toString('hex')],
    },
	ID: {
        propName: 'id',
        defaultValue: getRandomBytes(SIZE_BYTE_ID).toString('hex'),
    },
});

const mockOptionalProperties = (transaction) => {
    Object
    .values(OPTIONAL_TRANSACTION_PROPERTIES)
    .forEach(optionalPropInfo => {
        if (!(optionalPropInfo.propName in transaction))
            transaction[optionalPropInfo.propName] = transaction[optionalPropInfo.defaultValue];
    });

	return transaction;
};
```